examples/terraform/aws-dualstack/main.tf

```diff
@@ -0,0 +1,403 @@
+  ami                   = var.ami == "" ? data.aws_ami.ami.id : var.ami
+  zoneA                 = data.aws_availability_zones.available.names[0]
+  zoneB                 = data.aws_availability_zones.available.names[1]
+  zoneC                 = data.aws_availability_zones.available.names[2]
+  vpc_mask              = parseint(split("/", data.aws_vpc.selected.cidr_block)[1], 10)
+  subnet_total          = pow(2, var.subnets_cidr - local.vpc_mask)
+  subnet_newbits        = var.subnets_cidr - (32 - local.vpc_mask)
+  worker_os             = var.worker_os == "" ? var.ami_filters[var.os].worker_os : var.worker_os
+  worker_deploy_ssh_key = var.worker_deploy_ssh_key ? [aws_key_pair.deployer.public_key] : []
+  ssh_username          = var.ssh_username == "" ? var.ami_filters[var.os].ssh_username : var.ssh_username
+  bastion_user          = var.bastion_user == "" ? var.ami_filters[var.os].ssh_username : var.bastion_user
+  kubeapi_endpoint      = var.disable_kubeapi_loadbalancer ? aws_instance.control_plane.0.private_ip : aws_elb.control_plane.0.dns_name
+  loadbalancer_count    = var.disable_kubeapi_loadbalancer ? 0 : 1
+
+  initial_machinedeployment_spotinstances = var.initial_machinedeployment_spotinstances_max_price > 0
+
+  subnets = {
+    (local.zoneA) = length(aws_subnet.public.*.id) > 0 ? aws_subnet.public[0].id : ""
+    (local.zoneB) = length(aws_subnet.public.*.id) > 0 ? aws_subnet.public[1].id : ""
+    (local.zoneC) = length(aws_subnet.public.*.id) > 0 ? aws_subnet.public[2].id : ""
+  }
+}
+
+################################# DATA SOURCES #################################
+
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+data "aws_ami" "ami" {
+  most_recent = true
+  owners      = var.ami_filters[var.os].owners
+
+  filter {
+    name   = "name"
+    values = var.ami_filters[var.os].image_name
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+}
+
+data "aws_vpc" "selected" {
+  id = var.vpc_id
+}
+
+data "aws_internet_gateway" "default" {
+  filter {
+    name   = "attachment.vpc-id"
+    values = [data.aws_vpc.selected.id]
+  }
+}
+
+resource "random_integer" "cidr_block" {
+  min = 0
+  max = local.subnet_total - 1
+}
+
+############################### NETWORKING SETUP ###############################
+
+resource "aws_subnet" "public" {
+  count                   = 3
+  availability_zone       = data.aws_availability_zones.available.names[count.index]
+  map_public_ip_on_launch = true
+  vpc_id                  = data.aws_vpc.selected.id
+
+  cidr_block = cidrsubnet(
+    data.aws_vpc.selected.cidr_block,
+    local.subnet_newbits,
+    (random_integer.cidr_block.result + count.index) % local.subnet_total,
+  )
+
+  ipv6_cidr_block = cidrsubnet(
+    data.aws_vpc.selected.ipv6_cidr_block,
+    local.subnet_newbits,
+    (random_integer.cidr_block.result + count.index) % local.subnet_total,
```

Comment:
Is this actually going to work? `subnet_newbits` and `subnet_total` are based on IPv4 CIDRs. Don't we have a bit more addresses with IPv6?